pkg/api/openshiftcluster_validatedynamic.go

```diff
@@ -19,13 +19,18 @@ import (
 	"github.com/dgrijalva/jwt-go"
 	"k8s.io/apimachinery/pkg/util/wait"
 
-	"github.com/Azure/ARO-RP/pkg/api"
 	"github.com/Azure/ARO-RP/pkg/env"
 	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/authorization"
 	utilpermissions "github.com/Azure/ARO-RP/pkg/util/permissions"
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 )
 
+func NewOpenShiftClusterDynamicValidator(env env.Interface) OpenShiftClusterDynamicValidator {
```

Comment:
I am onboard with bringing the dynamic validator out of the versioned API if it helps us.  It seems like adding it into pkg/api causes a lot of churn elsewhere due to circular dependencies and we probably agree it's against the direction we want to go.  Does it help to put it into a separate package? (sub-package?)